packages/medusa/src/api/middlewares/process-cached-events.ts

```diff
@@ -0,0 +1,23 @@
+import { NextFunction, Request, Response } from "express"
+import { IEventBusService } from "../../interfaces/services/event-bus"
+
+export default (): ((
```

Comment:
self-review: Attach listener on the response, ensuring we always process the cached events. In case an error has been thrown leading to a transaction rollback, the cache invalidating middleware will have destroyed the cache as a preceding step.